sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.cs

```diff
@@ -4587,7 +4587,7 @@ bool IsAlwaysIncluded(Cursor cursor)
         bool IsExcludedByConfig(Cursor cursor)
         {
             return (_config.ExcludeFunctionsWithBody && (cursor is FunctionDecl functionDecl) && functionDecl.HasBody)
-                || (!_config.GenerateTemplateBindings && (cursor is TemplateDecl));
+                || (!_config.GenerateTemplateBindings && ((cursor is TemplateDecl) || (cursor is ClassTemplateSpecializationDecl)));
```

Comment:
Some other template related declarations that don't inherit from these two include:
* `FriendTemplateDecl`
* `NonTypeTemplateParmDecl`
* `TemplateParamObjectDecl`
* `TemplateTypeParmDecl`
* `VarTemplatePartialSpecializationDecl`
* `VarTemplateSpecializationDecl`

Of these, I think `VarTemplateSpecializationDecl` and `VarTemplatePartialSpecializationDecl` are the only two that don't require first being in an already skipped templated declaration context. But it may be important to look a bit more deeply.